pyproject.toml

```diff
@@ -44,6 +44,7 @@ pytest-cov = "~4.0.0"
 flaky = "~3.7.0"
 pytest-xdist = "~3.2.0"
 yamllint = "~1.31.0"
+typing-extensions = "~4.8.0"
```

Comment:
I'm not sure we should add this dependency only for `assert_nerver()`. I think we should keep doing `assert False` and we can change all these asserts after we deprecate python 3.10.

Reply:
You're right. Ok!

Reply:
I think it can only be considered a dev-dependency if it's not needed in runtime. But since `from typing_extensions import assert_never` is used directly in a module runtime would fail if `typing_extensions` is not installed. The only thing preventing it from failing in runtime is that some other runtime dependency pulls this lib. So I think we should put it in the `[tool.poetry.dependencies]` section.

Reply:
Makes sense, I got it mixed up there. I'll change this. Thanks!

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/830/commits/99af497b6300f65304ed647aa26b2562cb1a16bf